front/src/applications/stdcm/hooks/useLinkedPathSearch.ts

```diff
@@ -35,28 +39,28 @@ const useLinkedPathSearch = () => {
   const [hasSearchBeenLaunched, setHasSearchBeenLaunched] = useState(false);
   const [trainNameInput, setTrainNameInput] = useState('');
   const [linkedPathDate, setLinkedPathDate] = useState(selectableSlot.start);
-  const [linkedPathResults, setLinkedPathResults] = useState<StdcmLinkedPathResult[]>([]);
+  const [linkedPathResults, setLinkedPathResults] = useState<StdcmLinkedPathResult[]>();
 
-  const getExtremitiesDetails = useCallback(
-    async (pathItemList: PathItem[]) => {
-      const origin = pathItemList.at(0)!;
-      const destination = pathItemList.at(-1)!;
-      if (!('operational_point' in origin) || !('operational_point' in destination))
-        return undefined;
-      const originId = origin.operational_point;
-      const destinationId = destination.operational_point;
+  const getExtremityDetails = useCallback(
+    async (pathItem: PathItem) => {
+      if (!('operational_point' in pathItem) && !('uic' in pathItem)) return undefined;
+
+      const pathItemQuery =
+        'operational_point' in pathItem
+          ? ['=', ['obj_id'], pathItem.operational_point]
+          : ([
+              'and',
+              ['=', ['uic'], pathItem.uic],
+              ['=', ['ch'], pathItem.secondary_code],
+            ] as SearchQuery);
 
       try {
         const payloadOP = {
           object: 'operationalpoint',
-          query: ['or', ['=', ['obj_id'], originId], ['=', ['obj_id'], destinationId]],
-        };
-        const resultsOP = await postSearch({ searchPayload: payloadOP, pageSize: 25 }).unwrap();
-        const groupedResults = groupBy(resultsOP, 'obj_id');
-        return {
-          origin: groupedResults[originId][0],
-          destination: groupedResults[destinationId][0],
+          query: pathItemQuery,
         };
+        const opDetails = await postSearch({ searchPayload: payloadOP, pageSize: 25 }).unwrap();
```

Comment:
You can add a cast here 
```suggestion
        const opDetails = (await postSearch({
          searchPayload: payloadOP,
          pageSize: 25,
        }).unwrap()) as SearchResultItemOperationalPoint[];
```